language-plutus-core/src/Language/PlutusCore/Constant/Typed.hs

```diff
@@ -312,6 +340,15 @@ readDynamicBuiltinM
     => Term TyName Name () -> Evaluate m (Convert dyn)
 readDynamicBuiltinM term = withEvaluator $ \eval -> readDynamicBuiltin eval term
 
+-- | The denotation of a term whose type is a bound variable.
+-- I.e. the denotation of such a term is the term itself.
+-- This is because we have parametricity in Haskell, so we can't inspect a value whose
+-- type is a bound variable, so we never need to convert such a term from Plutus Core to Haskell
+-- and back and instead can keep it intact.
+newtype QuotedTerm (text :: Symbol) (unique :: Nat) = QuotedTerm
```

Comment:
I think `QuotedTerm` is a little confusing considering `Quote`. Perhaps `AbstractTerm`, or `OpaqueTerm`?

Reply:
`OpaqueTerm` is great, thanks, I'll change the name.